test/04_calendar_date_overrides.js

```diff
@@ -37,48 +37,170 @@ var Titles = Romcal.Titles;
       });
-      _.eq(date.key, 'saintIsidoreOfSevilleBishopAndDoctorOfTheChurch').should.be.ok();
+      _.size(date).should.be.eql(1);
     });
-    it('Therefore, national calendar of spain should only have one occurence of this feast on the 26th of April', function() {
-      var occurences = _.filter(spainDates, function(d) {
+    it('Therefore, national calendar of spain should only have one occurrence of this feast on the 26th of April', function() {
+      var occurrences = _.filter(spainDates, function(d) {
         return _.eq(d.key, 'saintIsidoreOfSevilleBishopAndDoctorOfTheChurch');
       });
-      _.size(occurences).should.be.eql(1);
+      _.size(occurrences).should.be.eql(1);
+    });
+  });
+
+  describe('Testing the priority option for celebrations', function() {
+    CalendarsDef.test = {};
+    CalendarsDef.test.dates = (year) => {
+      return [
+        {
+          "key": "maryMotherOfTheChurch",
+          "type": Types.OPT_MEMORIAL,
+          "moment": ( y => Dates.pentecostSunday( y ).add( 1, 'days'))( year ),
+          "data": {
+            "prioritized": true
+          }
+        },
+        {
+          "key": "ashWednesday",
+          "type": Types.SUNDAY,
+          "moment": Dates.ashWednesday(year)
+        },
+        {
+          "key": "saintLukeTheEvangelist",
+          "type": Types.COMMEMORATION,
+          "moment": moment.utc({year: year, month: 9, day: 18}),
+          "data": {
+            "prioritized": true
+          },
+        },
+        {
+          "key": "aSampleCelebration1",
+          "type": Types.MEMORIAL,
+          "moment": moment.utc({year: year, month: 10, day: 9}),
+          "data": {
+            "prioritized": true
+          }
+        },
+        {
+          "key": "aSampleCelebration2",
+          "type": Types.SOLEMNITY,
+          "moment": moment.utc({ year: year, month: 11, day: 25 }),
+          "data": {
+            "prioritized": true
+          }
+        }
+      ];
+    };
+
+    var year = 2020;
+    var testDates = Calendar.calendarFor({
+      country: 'test',
+      year: year
+    });
+
+    it('A celebration with a higher type and the same key cannot replace an existing prioritized celebration', function() {
+      var dates = _.filter(testDates, function(d) {
+        return _.eq(d.key, 'ashWednesday');
+      });
+      _.size(dates).should.be.eql(1);
+      _.head(dates).type.should.be.eql(Types.FERIA);
+    });
+    it('A new prioritized celebration will replace any existing non-prioritized celebration', function() {
+      var dates = _.filter(testDates, function(d) {
+        return _.eq(d.key, 'saintLukeTheEvangelist');
+      });
+      _.size(dates).should.be.eql(1);
+      _.head(dates).type.should.be.eql(Types.COMMEMORATION);
+    });
+    it('An existing and prioritized celebration can be replaced by a new prioritized celebration having the same key (whatever its type rank)', function() {
```

Comment:
`whatever its type rank` → `whatever is its rank`

Reply:
it's important to keep the keyword `type` since it correspond to a specific item property, and because there is different kind of ranks managed by romcal (date, types, priority...)

Reply:
Well, I agree it is important, but _rank_ is used throughout [_General Norms for the Liturgical Year and the Calendar_](http://www.romcal.net/norms.html), therefore I don’t really like the term `type`. I’d even suggest to rename `Types.js` to `Ranks.js` (just I suggested to rename _national_ calendars to _particular_ ones).

> there is different kind of ranks managed by romcal (date, types, priority...)

I don’t really get this. Why would you say the a date or a priority would be a _rank_? They are attributes of a celebration, aren’t they?

Reply:
Because romcal manage all the celebrations during a year as a big array of DateItems, so they need to be ordered by date, and then for each days, by different ranking properties: if a celebration is prioritized or not, by its rank type, from where this celebration has been defined (from general, national...), etc.

Reply:
Okay, do as you wish (in the codebase comments). Still I think that the _type_ (as in solemnity or feast) should be renamed to _rank_, but that is out of scope of this PR.